### PR TITLE
make hubot respond to “lunch at ..” and “lunch me” commands

### PR DESCRIPTION
**Pull request by guncha**
For all your lunch scheduling needs!

### CODE REVIEW

**Comment by chanks** [comment]
:+1: functionality needed urgently.

File: scripts/lunch.coffee [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#   None
+#
+# Commands:
+#   <botname> lunch at [location]
+#   <botname> lunch me
+
+makeRecord = (now = new Date()) ->
+
+  return {
+    last_lunch: now.getFullYear() + "-" + (now.getMonth() + 1) + "-" + now.getDate()
```
Comment (NevilleS) [inline scripts/lunch.coffee:17]: I think that storing the last timestamp twice (once as a formatted string) smells a bit funny. This makes more sense as just presentation logic, doesn't really need to be cached...

File: scripts/lunch.coffee [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    if record = robot.brain.data.lunch_spots[location]
+      msg.send "last lunch at #{location} was #{record.last_lunch}"
+    else
+      msg.send "have a great lunch!"
+
+    robot.brain.data.lunch_spots[location] = makeRecord()
+
+
+  robot.respond /lunch seed$/i, (msg) ->
```
Comment (NevilleS) [inline scripts/lunch.coffee:36]: While good for testing, I don't think we should check this in; too easy to accidentally wipe out the valuable lunch database...! :smile:

File: scripts/lunch.coffee [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    console.log robot.brain.data.lunch_spots
+
+  robot.respond /lunch me$/i, (msg) ->
+
+    candidates = []
+    beforeTimestamp = (new Date()).getTime() - 7 * 24 * 60 * 60 * 1000
+
+    for location, record of robot.brain.data.lunch_spots
+      candidates.push [location, record] if record.timestamp < beforeTimestamp
```
Comment (NevilleS) [inline scripts/lunch.coffee:62]: The candidates array would be cleaner as just an array of locations, forget the record part

File: scripts/lunch.coffee [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+module.exports = (robot) ->
+
+  robot.respond /lunch at (.+)$/i, (msg) ->
+
+    location = msg.match[1]
+    robot.brain.data.lunch_spots ||= {}
+
+    if record = robot.brain.data.lunch_spots[location]
+      msg.send "last lunch at #{location} was #{record.last_lunch}"
```
Comment (NevilleS) [inline scripts/lunch.coffee:29]: I feel like it's non-obvious that it "worked" in this case. Should probably be like:

> pgbot lunch at chipotle
> OK, enjoy lunch at chipotle! You were there last on 1-12-2015

And...

> pgbot lunch at eleven madison park
> OK, eleven madison park for lunch! That looks like a new place - hope it's good...

**Comment by NevilleS** [comment]
Some minor comments. Also I think it probably needs a bit more personality, I think I'll probably add some random responses in again (e.g. "Potbelly again, why am I not surprised?") :+1:

**Comment by guncha** [comment]
Neville, I was going to make some changes to it over the weekend / next
week. I'll just create a new PR.

## 

Best,
G.

**Comment by NevilleS** [comment]
Haha, I was going to tweak it today, but I'll let you do that instead :+1:

**Pull request merged**